src/Umbraco.Core/Services/ContentService.cs

```diff
@@ -1131,6 +1131,9 @@ public void MoveToRecycleBin(IContent content, int userId = 0)
                     return;
                 }
 
+                // Add a relation for the item being deleted, so that we can know the original parent for if we need to restore later
+                ApplicationContext.Current.Services.RelationService.Relate(content.Parent(), content, Constants.Conventions.Relations.RelateParentDocumentOnDelete);

```

Comment:
Thanks @sitereactor I wasn't aware of the `RelateOnCopyHandler` class.
I'll update accordingly.